File: sdsd_book/samples/csharp-dotnet/SdsdSample.Core/AuthService.cs
using System;

namespace SdsdSample.Core
{
    public class AuthService
    {
        public bool Authenticate(string tenantId, string username, string password)
        {
            // SDSD Invariant: Tenant Isolation
            if (string.IsNullOrWhiteSpace(tenantId))
            {
                throw new ArgumentException("Tenant ID is required for isolation.", nameof(tenantId));
            }
            
            // Dummy logic
            if (username == "admin" && password == "secure123")
            {
                return true;
            }
            return false;
        }
    }
}

File: sdsd_book/samples/python-pytest/src/auth_service.py
class AuthService:
    def authenticate(self, tenant_id: str, username: str, password: str) -> bool:
        # SDSD Invariant: Tenant Isolation
        if not tenant_id or not tenant_id.strip():
            raise ValueError("Tenant ID is required for isolation.")
            
        # Dummy logic
        if username == "admin" and password == "secure123":
            return True
        return False
